sampleinput.yaml

```diff
@@ -0,0 +1,14 @@
+creds:
+  github:
+    token: my_token
+  gitlab:
+    token: my_token
+    host: my_hostname
+git:
+   - github:
+     - user_name/repo_name
+     - user_name
+   - gitlab:
```

Comment:
One thing worth considering: what if there is more than one instance of gitlab or github, etc? This structure doesn't allow for that flexibility. I feel like gitlab/github/etc. are types and you need to assign unique names to different instances.